mlir/include/mlir/IR/DialectBase.td

```diff
@@ -34,6 +34,8 @@ class Dialect {
   // pattern or interfaces.
   list<string> dependentDialects = [];
 
+  dag discardableAttrs = (ins);
```

Comment:
Is my mental model correct that these are all effectively optional attributes for which there are helpers generated? But they are different than the others as not verified, most likely empty? (e.g., I'm not sure if doing this vs making these optional attributes when one would select which ...)

Reply:
I don't know if I followed correctly what you're asking about, but these are helpers to manage discardable attributes. The idea is that for example the ROCDL dialect involves a discardable attribute with the key `rocdl.reqd_work_group_size`, and the associated Attribute must be a `DenseI32ArrayAttr`.
This feature will generate a helper for managing this discardable attribute on operations. The operations don't know anything about the attribute, it's a property of the dialect itself.

Reply:
That makes sense to me. These being dialect known discardable rather than op known optional.